website/docs/provider_type.mdx

```diff
@@ -0,0 +1,34 @@
+---
+title: Provider Type
+---
+
+### \[Sync/Async]\[Stateless/Stateful]Provider
+
+- <i><u>Note:</u></i> This is just a way of thinking, not actual types. You don’t need to pick
+  a type when using code-generation variant anyway!
+
+- `Sync/Async` is being decided by your return type, either of your function or the [build] method
+  in case of using a class (stateful):
+   - `Sync` ⇒ T `Async` ⇒ Future<T\> or Stream<T\>
+
+- `Stateless/Stateful` is being decided by whether you need to expose methods to modify the state or not (side-effects):
+   - `Stateless` ⇒ use a function. `Stateful` ⇒ use a class with [build] method + side effect methods.
+
+
+### For non code-generation variant:
+You have to manually determine the type of your provider:
+
+- Sync Stateless Provider ⇒ [Provider]
```

Comment:
I don't really want to have folks think about the internal "provider type" being used. That's the old way of thinking.

The generated code is an implementation detail and could very well change at any time to not generate a StreamProvider when returning a Stream in a function for instance.   
In fact it's going to change pretty soon.

Reply:
I totally agree. but, currently the `non code-generation` variant is an option and not deprecated yet, right?

I think clarifying the relation between `code-generation`'s way of thinking and `non code-generation`'s actual types might be helpful, until `non code-generation` variant get deprecated and that means `All Providers` section will be removed/updated too?

Reply:
I find that `non code-generation variant` has 2 benefits:

1. Discourage folks from using it as it clearly announce that they've to `manually determine the type`.
2. Encourage folks that already using `non code-generation` to migrate into code-gen easily. i.e: checking i'm using `FutureProvider` then I'd migrate to `function` with an `async` return type.

Anyway, It's up to you :)

Reply:
I probably would move this to the "about code-generation" page then.